### PR TITLE
Add `Backup` resource custom hooks, terminalCodes and e2e tests

### PR DESCRIPTION
**Pull request by A-Hilaly**
Part of https://github.com/aws-controllers-k8s/community/issues/803

Description of changes:
- Add custom hooks, terminalCodes to `Backup` resource in `generator.yaml`
- Add e2e tests for `Backup` create and delete operations

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by A-Hilaly** [comment]
Note to reviewers: This PR needs to be rebased once #3 and #4 are merged
/hold

File: pkg/resource/backup/hooks.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+	ackrequeue "github.com/aws-controllers-k8s/runtime/pkg/requeue"
+
+	"github.com/aws-controllers-k8s/dynamodb-controller/apis/v1alpha1"
+)
+
+var (
+	// TerminalStatuses are the status strings that are terminal states for a
+	// backup.
+	TerminalStatuses = []v1alpha1.BackupStatus_SDK{}
```
Comment (jaypipes) [inline pkg/resource/backup/hooks.go:27]: this is meant to be empty?

Comment (A-Hilaly) [reply pkg/resource/backup/hooks.go:27]: There are no terminal statuses for `Backup`, for now.. https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BackupDetails.html

File: test/e2e/tests/test_backup.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+        except Exception as e:
+            logging.debug(e)
+            return None
+
+    def backup_exists(self, dynamodb_client, backup_arn: str) -> bool:
+        return self.get_backup(dynamodb_client, backup_arn) is not None
+
+    def test_smoke(self, dynamodb_client, dynamodb_table):
+        pytest.skip(f"No tests for Backup")
```
Comment (RedbackThomson) [inline test/e2e/tests/test_backup.py:96]: Oops

**Review by RedbackThomson — APPROVED** [review]

File: pkg/resource/backup/hooks.go (hunk 1 above)
Comment (jaypipes) [reply pkg/resource/backup/hooks.go:27]: k.

File: test/e2e/tests/test_backup.py (hunk 2 above)
Comment (A-Hilaly) [reply test/e2e/tests/test_backup.py]: Nice catch!

**Comment by A-Hilaly** [comment]
Mmmmm it looks like tests are randomly failing from time to time.. investigating that.
/hold

**Review by jaypipes — APPROVED** [review]
tres bien, merci!

**Pull request merged**